src/controllers/students.controller.js

```diff
@@ -0,0 +1,32 @@
+const httpStatus = require('http-status');
+const ApiError = require('../utils/ApiError');
+const catchAsync = require('../utils/catchAsync');
+const { createStudentNotification } = require('../utils/notifications');
+const { userService, tokenService, notificationService } = require('../services');
+
+const createStudent = catchAsync(async (req, res) => {
+    const student = await userService.createUser(req.body);
+    const token = await tokenService.generateAuthTokens(student);
+    const receiverNotification = createStudentNotification(token);
+    await notificationService.sendNotification(messageBody.to, receiverNotification);
+    res.send(student);
+});
+
+const getStudent = catchAsync(async (req, res) => {
+    const student = await userService.getUserById(req.params.studentId);
```

Comment:
have you checked if it is student herself/himself or his class's teacher try to call this function